src/software/jetson_nano/services/motor.cpp

```diff
@@ -102,54 +111,223 @@ MotorService::MotorService(const RobotConstants_t& robot_constants,
+    int gstat = tmc6100_readInt(motor, TMC6100_GSTAT);
+    std::bitset<32> gstat_bitset(gstat);
 
-    switch (direct_control.wheel_control_case())
+    CHECK(gstat_bitset.any() == false)
+        << "TMC6100 offline, no fault bits detected."
+        << " The reset bit should be set. This could indicate that the driver isn't"
+        << " communicating over SPI or the reset pin is not connected.";
+
+    if (gstat_bitset.any())
     {
-        case TbotsProto::DirectControlPrimitive::WheelControlCase::kDirectPerWheelControl:
-        {
-            // TODO (#2456) until we figure out the right factor considering
-            // the gear ratio
-            static float RANDOM_SCALING_FACTOR = 100.0;
+        LOG(WARNING) << "======= Faults For Motor " << std::to_string(motor)
+                     << "=========";
+    }
+
+    if (gstat_bitset[0])
+    {
+        LOG(WARNING)
+            << "Indicates that the IC has been reset. All registers have been cleared to reset values."
+            << "Attention: DRV_EN must be high to allow clearing reset";
+    }
+
+    if (gstat_bitset[1])
+    {
+        LOG(WARNING)
+            << "drv_otpw : Indicates, that the driver temperature has exceeded overtemperature prewarning-level."
+            << "No action is taken. This flag is latched.";
+    }
+
+    if (gstat_bitset[2])
+    {
+        LOG(WARNING)
+            << "drv_ot: Indicates, that the driver has been shut down due to overtemperature."
+            << "This flag can only be cleared when the temperature is below the limit again."
+            << "It is latched for information.";
+    }
+
+    if (gstat_bitset[3])
+    {
+        LOG(WARNING) << "uv_cp: Indicates an undervoltage on the charge pump."
+                     << "The driver is disabled during undervoltage."
+                     << "This flag is latched for information.";
+    }
+
+    if (gstat_bitset[4])
+    {
+        LOG(WARNING) << "shortdet_u: Short to GND detected on phase U."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
+
+    if (gstat_bitset[5])
+    {
+        LOG(WARNING) << "s2gu: Short to GND detected on phase U."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
 
-            // TODO (#2456) We can only spin 1 motor right now, figure out how
-            // to spin the rest and update the following code
+    if (gstat_bitset[6])
+    {
+        LOG(WARNING) << "s2vsu: Short to VS detected on phase U."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
+
+    if (gstat_bitset[8])
+    {
+        LOG(WARNING) << "shortdet_v: V short counter has triggered at least once.";
+    }
+
+    if (gstat_bitset[9])
+    {
+        LOG(WARNING) << "s2gv: Short to GND detected on phase V."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
```

Comment:
looks like a map from index to warning message could be used, then we loop through the gstat_bitset and then lookup the warning message

Reply:
just to make sure I understand how you are thinking the map would work, we would turn the bitset into a vector and find all positions of 1, then loop over and lookup in map?

Reply:
it would map `9` to `"s2gv: Short to GND detected on phase V. "The driver becomes disabled until flag becomes cleared."`

Reply:
i get the mapping part, how would the lookup work? is there a clean way to extract indexes that are set to 1 in a bitset, I can't think of any/find anything online

Reply:
Im not super familiar with bitset, but cant you not use bitwise operators to loop through the index? something like
```cpp
    std::bitset<32> gstat_bitset;
    ...
    // index -> msg
    std::map<int, std::string> index_map;
    for (int a = 1, index = 0; index < 32; a << 1, index++)
    {
        if (gstat_bitset[index] & a)
        {
            LOG(WARNING) << index_map[index];
        }
    }
```

Reply:
Another approach which is slightly different from what you have and I've seen elsewhere is using defined flags instead of a bitset and indices, so e.g.
```cpp
    unsigned int ic_reset_error_flag = 1 << 0;
    unsigned int drv_otpw_error_flag = 1 << 1;
    ...
    if (gstat & ic_reset_error_flag)
    {
        ...
    }

    if (gstat & drv_otpw_error_flag)
    {
        ...
    }
    ...
```
Though I think this is more common when you wanna check if two flags are set at the same time.

Reply:
I am wondering if there is a drawback to having it laid out how it currently is? I personally find it simpler/more readable than doing bit manipulation to extract the index and do a lookup

I like the second implementation but its also very similar to the bitset impl that's currently in the PR